### DISCUSSION TITLE
Adding Constraints

### DISCUSSION THREAD

**Discussion by SiruiXie** [tesseract-robotics/tesseract_planning#569] [General]
Great work @Levi-Armstrong ! Really helped me a lot. 
I was trying to add some constraint to the optimization process, and by default, not using `ifopt`, but I find it hard to understand where should be the interface to add that constraint, take `basic_cartesian_example` for instance. Also, I was wondering if there exists an interface to add any constrains between different frames, or time_steps, in a trajectory.
It would be appreciated if I could get any reply on this issue. Thanks a lot!

**Comment by Levi-Armstrong** [comment]
This depends on if you are wanting to create new costs/constraints or use existing ones. If wanting to leverage existing ones these are done by configuring the trajopt profile. If you wanted to add new one you will be better off looking at the trajopt repository and its existing costs and constraints.
